m3u8/reader.go

```diff
@@ -344,6 +344,34 @@ func decodeLineOfMasterPlaylist(p *MasterPlaylist, state *decodingState, line st
 		state.variant = variant
 		state.variant.Iframe = true
 		p.Variants = append(p.Variants, state.variant)
+	case strings.HasPrefix(line, "#EXT-X-DEFINE:"): // Define tag
+		var (
+			name       string
+			value      string
+			defineType DefineType
+		)
+
+		switch {
+		case strings.HasPrefix(line, "#EXT-X-DEFINE:NAME="):
+			defineType = VALUE
+			_, err = fmt.Sscanf(line, "#EXT-X-DEFINE:NAME=%q,VALUE=%q", &name, &value)
+		case strings.HasPrefix(line, "#EXT-X-DEFINE:QUERYPARAM="):
+			defineType = QUERYPARAM
+			_, err = fmt.Sscanf(line, "#EXT-X-DEFINE:QUERYPARAM=%q", &name)
+		case strings.HasPrefix(line, "#EXT-X-DEFINE:IMPORT="):
+			return fmt.Errorf("EXT-X-DEFINE IMPORT not allowed in master playlist")
```

Comment:
Good to report an error on this.

(Some linting tools want error messages always start with a lower-case letter, but here it makes sense to use upper case).